pkg/innerring/processors/governance/process_update.go

```diff
@@ -0,0 +1,98 @@
+package governance
+
+import (
+	"encoding/binary"
+	"sort"
+
+	"github.com/nspcc-dev/neofs-node/pkg/innerring/invoke"
+	"go.uber.org/zap"
+)
+
+const alphabetUpdateIDPrefix = "AlphabetUpdate"
+
+func (gp *Processor) processAlphabetSync() {
+	if !gp.alphabetState.IsAlphabet() {
+		gp.log.Info("non alphabet mode, ignore alphabet sync")
+		return
+	}
+
+	mainnetAlphabet, err := gp.mainnetClient.NeoFSAlphabetList()
+	if err != nil {
+		gp.log.Error("can't fetch alphabet list from main net",
+			zap.String("error", err.Error()))
+		return
+	}
+
+	sidechainAlphabet, err := gp.morphClient.Committee()
+	if err != nil {
+		gp.log.Error("can't fetch alphabet list from side chain",
+			zap.String("error", err.Error()))
+		return
+	}
+
+	newAlphabet, err := newAlphabetList(sidechainAlphabet, mainnetAlphabet)
+	if err != nil {
+		gp.log.Error("can't merge alphabet lists from main net and side chain",
+			zap.String("error", err.Error()))
+		return
+	}
+
+	if newAlphabet == nil {
+		gp.log.Info("no governance update, alphabet list has not been changed")
+		return
+	}
+
+	gp.log.Info("alphabet list has been changed, starting update")
+	sort.Sort(newAlphabet)
+
+	// 1. Vote to side chain committee via alphabet contracts.
+	err = gp.voter.VoteForSidechainValidator(newAlphabet)
+	if err != nil {
+		gp.log.Error("can't vote for side chain committee",
+			zap.String("error", err.Error()))
+	}
+
+	// 2. Update NeoFSAlphabet role in side chain.
+	innerRing, err := gp.morphClient.NeoFSAlphabetList()
+	if err != nil {
+		gp.log.Error("can't fetch inner ring list from side chain",
+			zap.String("error", err.Error()))
+	} else {
+		newInnerRing, err := updateInnerRing(innerRing, sidechainAlphabet, newAlphabet)
+		if err != nil {
+			gp.log.Error("can't create new inner ring list with new alphabet keys",
+				zap.String("error", err.Error()))
+		}
+
+		sort.Sort(newInnerRing)
```

Comment:
1. can `newInnerRing` have the same items as `innerRing`, and if so, should we call `UpdateNeoFSAlphabetList` in this case?
2. if `err != nil` in previous statement should we perform this action or it must be placed in `else` block?

Reply:
1. It is not possible. In this case `mainnet` should be equal to `sidechain`, but `newAlphabetList` will return `nil`  and it will be processed:
```go
	if newAlphabet == nil {
		gp.log.Info("no governance update, alphabet list has not been changed")
		return
	}
```

2. You are right about this, it should placed in else block. Fixed.